ewfs/scenario.py

```diff
@@ -134,16 +135,20 @@ def circuit(self) -> QuantumCircuit:
 
     def _initialize_circuit(self) -> QuantumCircuit:
         """Initialize the classical measurement registers based on the strategy."""
-        if self.friend_state is CNOT_LADDER:
-            if self.strategy is MAJORITY_VOTE:
-                if self.alice_setting is PEEK and self.bob_setting is not PEEK:
-                    measurement = ClassicalRegister(self.charlie_size + 1, name="measurement")
-                else:
-                    measurement = ClassicalRegister(self.meas_size, name="measurement")
-
-            elif self.strategy is RANDOM:
+        if self.strategy is MAJORITY_VOTE:
+            if self.alice_setting is PEEK and self.bob_setting is PEEK:
+                measurement = ClassicalRegister(self.charlie_size + self.debbie_size, name="measurement")
+            elif self.alice_setting is PEEK and self.bob_setting is not PEEK:
+                measurement = ClassicalRegister(self.charlie_size + 1, name="measurement")
+            elif self.alice_setting is not PEEK and self.bob_setting is PEEK:
+                measurement = ClassicalRegister(self.debbie_size + 1, name="measurement")
+            else:
                 measurement = ClassicalRegister(self.sys_size, name="measurement")
 
+        elif self.strategy is RANDOM:
+            measurement = ClassicalRegister(self.sys_size, name="measurement")
```

Comment:
[nit]: We could do something a bit more streamlined:

```suggestion
if self.strategy is MAJORITY_VOTE:
    size = (
        self.charlie_size + self.debbie_size if self.alice_setting is PEEK and self.bob_setting is PEEK
        else self.charlie_size + 1 if self.alice_setting is PEEK
        else self.debbie_size + 1 if self.bob_setting is PEEK
        else self.sys_size
    )
elif self.strategy is RANDOM:
    size = self.sys_size

measurement = ClassicalRegister(size, name="measurement")
```